### ISSUE TITLE
Add vmId to VirtualMachine

### ISSUE THREAD

**Issue by jianghaolu** [Azure/azure-rest-api-specs#188]
Per customer request: https://github.com/Azure/azure-sdk-for-java/pull/619

The vmId field is available via REST - but it is not part of the SDK, to be more precise, the VirtualMachine object does not have a String vmId member, so when the REST call is returned, though available via REST, this value is not present on the virtual machine.
I need this value in order to uniquely identify this virtualMachine object - and I cannot use the id value since it is based on the vm's name.

@noaml2gitHub

**Comment by noaml2gitHub** [comment]
@jianghaolu - thanks for the help.
I really hope this gets fixed ASAP.

Thanks again

**Comment by danielestevez** [comment]
Hi @noaml2gitHub @jianghaolu 
I also asked for this in Azure Feedback forums (long time ago!) [https://feedback.azure.com/forums/216843-virtual-machines/suggestions/6750343-retrieve-vm-internaluuid-from-rest-api](https://feedback.azure.com/forums/216843-virtual-machines/suggestions/6750343-retrieve-vm-internaluuid-from-rest-api)
Let's hope it gets more interest now.

**Comment by devigned** [comment]
@hyonholee can you provide an update on this issue?

**Comment by noaml2gitHub** [comment]
@hyonholee  - Can you **please** update here if/when you're planning of adding this?

**Comment by noaml2gitHub** [comment]
Can someone from the Azure team please comment here about my request?

**Comment by amarzavery** [comment]
@huangpf can you take a look at this?

Thanks,
Amar Zavery

On Apr 7, 2016, at 1:35 AM, "noaml2gitHub" <notifications@github.com<mailto:notifications@github.com>> wrote:


Can someone from the Azure team please comment here about my request?

-

**Comment by huangpf** [comment]
OK. Got a sample here. 
```
{
  "properties": {
    "vmId": "cbd1d46f-7336-45ff-9a04-2a020b5a228c",
    "hardwareProfile": {
      "vmSize": "Standard_A2"
    },
...
```

@singhkay @mahthi This is a gap to fill between API and spec (for SDKs in all languages). Please help triage. Thanks.

**Comment by devigned** [comment]
I believe this can be closed since #230 has been merged.

**Comment by noaml2gitHub** [comment]
Thanks for the reply, so when will it be available in the SDK? I saw that the released version is 0.9.3 and I still didn't see this code in 0.9.4-SNAPSHOT

**Comment by huangpf** [comment]
@jianghaolu Please help follow up with the Java SDK. Thanks.

**Comment by jianghaolu** [comment]
@noaml2gitHub This is added since 1.0.0-beta2 release: https://github.com/Azure/azure-sdk-for-java/blob/v1.0.0-beta3/azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java#L211. 

0.9.x SDKs are not generated from the swagger specs. A fix here will have no effect there.
Also, this is available in 2016-03-30 api version, while 0.9.x uses 2015-06-15.
